pkg/virt-api/rest/subresource.go

```diff
@@ -288,6 +281,28 @@ func (app *SubresourceAPIApp) StartVMRequestHandler(request *restful.Request, re
 		return
 	}
 
+	vmi, err := app.VirtCli.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
+
+	var (
+		vmiStatusPhase = v1.Unknown
+		vmiIsFinal     = false
+	)
+	if vmi != nil {
+		vmiStatusPhase = vmi.Status.Phase
+		vmiIsFinal = vmi.IsFinal()
+	}
+	if err != nil {
+		if !errors.IsNotFound(err) {
```

Comment:
maybe a bit matter of taste, but the error handling is getting a bit far away from the method execution... maybe this is a bit nicer?
```
vmiStatusPhase := v1.Unknown
vmiIsFinal := false

vmi, err := app.VirtCli.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
if err != nil {
	if !errors.IsNotFound(err) {
		response.WriteError(http.StatusInternalServerError, err)
		return
	}
} else {
	vmiStatusPhase = vmi.Status.Phase
	vmiIsFinal = vmi.IsFinal()
}
```